qiskit_nature/second_q/properties/angular_momentum.py

```diff
@@ -49,20 +52,53 @@ class AngularMomentum:
 
     Attributes:
         num_spatial_orbitals (int): the number of spatial orbitals.
-        overlap (np.ndarray | None): the overlap-matrix between the $\alpha$- and $\beta$-spin
-            orbitals. When this is `None`, the overlap-matrix is assumed to be identity.
     """
 
     def __init__(self, num_spatial_orbitals: int, overlap: np.ndarray | None = None) -> None:
         r"""
         Args:
             num_spatial_orbitals: the number of spatial orbitals in the system.
             overlap: the overlap-matrix between the $\alpha$- and $\beta$-spin orbitals. When this
-                is `None`, the overlap-matrix is assumed to be identity.
+                is ``None``, the overlap-matrix is assumed to be identity.
         """
         self.num_spatial_orbitals = num_spatial_orbitals
+        self._overlap: np.ndarray | None = None
         self.overlap = overlap
```

Comment:
For my education: isn't this overridden by the `overlap` property?

Reply:
I see it now (I think! 😺) this line will call the setter and apply the initialization logic. But then is there a point to type-hint `self._overlap` a union type? Couldn't it be just `np.ndarray`?

Reply:
The first line here makes sure that pylint does not complain about the private `_overlap` attribute not being defined inside the `__init__` (because otherwise it would be defined inside the `property` method.
I am typing it for the sake of completeness :+1: